### ISSUE TITLE
Strange client.properties issue that disconnected all people

### ISSUE THREAD

**Issue by benbaptist** [benbaptist/minecraft-wrapper#190]
```
Last packet IDs (Server->Client) before disconnection:
[('0x2', 47)]
Error in the Server->Client method:
Traceback (most recent call last):
  File "Wrapper.py/proxy.py", line 906, in handle
    if self.parse(id, original) and self.safe:
  File "Wrapper.py/proxy.py", line 853, in parse
    properties = client.properties
AttributeError: 'UUID' object has no attribute 'properties'
```

**Comment by suresttexas00** [comment]
What caused this?

Is this the code?
```
       if id == 0x38:
            head = self.read("varint:action|varint:length")
            z = 0
            while z < head["length"]:
                serverUUID = self.read("uuid:uuid")["uuid"]
                client = self.client.proxy.getClientByServerUUID(serverUUID)
                try:
                    uuid = client.uuid
                except:
                    uuid = client
                    z += 1
                if not client:
                    z += 1
                    continue
                z += 1
                if head["action"] == 0:
                    properties = client.properties
                    raw = ""
                    for i in properties:
                        raw += self.client.packet.send_string(i["name"])  # name
                        raw += self.client.packet.send_string(i["value"])  # value
```

I read up on stack exchange regarding use of init in classes and variables inside and outside the class init.
1) I noted you are using a variable name (properties) in your iterator that shadows the client class's `__init__` variable you are trying to access...  Not for sure if that is a problem for execution or readability or not...  
2) You are accessing the `.properties` variable that is defined inside the `__init__`.  I'm not sure you can actually access the `properties` variable using <class>.<variable> and come up with the correct value.  I saw an example as follows:

```
class MyClass(object):
     i = 123
     def __init__(self):
         self.i = 345

a = MyClass()
print a.i
345
print MyClass.i
123
```

**Comment by suresttexas00** [comment]
Deleted that previous comment because I am so confused at this point because you also created an object named the same as the class it called:

`client` = self.`client`.proxy.getClientByServerUUID(serverUUID)

**Comment by benbaptist** [comment]
yeah, I'm not sure what was going on there. I just quickly threw the error on here so that I could come back and fix it, but I don't really know exactly what led up to it.

**Comment by suresttexas00** [comment]
Are you going to consider the PEP-8 pull  request before you start changing more code?

**Comment by suresttexas00** [comment]
I have seen that error a lot.

**Comment by benbaptist** [comment]
I'll have to review it a bit more, but one thing that I'll admit that I'm 100% stubborn about is tabs vs spaces. I hate spaces so much for indentation, and the vast majority of arguments for "why you shouldn't use tabs" are easily reversible to "why you shouldn't use spaces". 

e.g. "You shouldn't mix tabs and spaces" - okay, that doesn't mean I should use spaces instead, that just tells me that I shouldn't mix them. 

"Tabs don't have a defined width" - the spaces method aren't either. you can have one space, two spaces, three spaces, four spaces, five spaces, etc. etc. etc.

My arguments against spaces:
- Pressing the space bar four times sucks. Some editors will convert tabs-to-spaces, but they aren't plentiful and my arguments against this are below.
- Making uber-quick edits in nano or another simplistic editor aren't as easy, for the aforementioned reason. I like to be able to open my code up in any editor that I have available (if I'm not at my workstation, for example) and do what I need with minimal hassle. 
- I've never seen an editor that hasn't had the ability to adjust tab width, besides maybe crap like Microsoft Notepad. However, I've seen editors that lack the ability to convert tabs to spaces easily. 
- PEP-8 is just a standard for Python's internal module code - there's no reason to follow it. 
- ME IS STUBBORN

I've seriously never had an issue with tabs, and spaces is just more hassle, so I'd really rather not go that route. The other stuff, on the other hand, I'll have to evaluate.

**Comment by suresttexas00** [comment]
I was of the same mindset, but, after looking into it, I only had one objection to spaces over tabs.. they use four bytes (ASCII) instead of one! ... LOL ... as though encryption and large disks don't exist!  Seriously, that is the only reason for tabs over spaces that I found.. A mental block of my own stubbornness regarding potential file size!
1) pressing space bar 4 times is an editor issue.  Most editors can "EASILY" be told that tab is four spaces.
2) Sorry... again about editor issues... But even in those editors, they are fixed width fonts that should minimize problems.
3) Tab width adjustment IS the EXACT problem that PEP-8 is addressing.  It is because of the ability to adjust tab widths that mixing tabs between different people's editors and spaces is prohibited.
4) I grant that as an open source language, you can't "make" people use PEP-8, but It looks nice, makes everyone's code style and readability consistent... and avoids errors.  Since it was essentially written by the Python's author, Guido VanRosswum, I think it pays to consider.
5) Sure, I get that...  but I think spaces will win just because some indentation cannot be done with just tabs (meaning you will have to mix potentially-adjustable-width tabs with spaces)  For instance (I know this is a bad example with unreal code, but..):
```
try (this_is_one_thing and
     that_is_another_thing):
     do_something()
```
Is impossible to indent that way using a 4-space tab.  It can be done with 5 spaces, though... and this was the chief reason I capitulated over my own stubbornness about file size...........  because of my insistence on using tabs, I now had to violate another PEP-8 that I had no problem with.

**Comment by suresttexas00** [comment]
I just started using PyCharm and it is a really nice tool.  Must nicer than MS visual studio, which was not able to resolve some import references, etc.. plus did not have all this code advice and error checking.

**Comment by suresttexas00** [comment]
Seriously, after using them, I don't miss the tabs anymore!

**Comment by suresttexas00** [comment]
Here's one I did not know and PyCharm did not flag these...
"Don't compare boolean values to True or False using == .

Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
"
I think I did a lot of `if greeting is True`....

**Comment by suresttexas00** [comment]
I am still finding this error on my server too.  This time I was prevented from logging in.  The only fix is to restart wrapper.

```
[12:02:36] [Wrapper.py/INFO] SurestTexas00 logged in (UUID: 4f953255-6775-4dc6-a612-fb42
30588eff | IP: 68.50.227.172)
[12:02:37] [Server thread/INFO]: SurestTexas00[/127.0.0.1:64625] logged in with entity i
d 27659 at (1593.3964577777137, 69.0, 464.64614363203304)
Error in the Server->Client method:
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 969, in handle
    if self.parse(id, original) and self.safe:
  File "Wrapper\proxy.py", line 916, in parse
    properties = client.properties
AttributeError: 'UUID' object has no attribute 'properties'

Last packet IDs (Server->Client) before disconnection:
[('0x39', 10), ('0x9', 2), ('0x37', 52), ('0x3e', 56), ('0x3e', 62), ('0x3e'[12:02:37] [
Server thread/INFO]: ºbSurestTexas00ºr joined the game,
63), ('0x3e', 48), ('0x3e', 61), ('0x38', 37), ('0x38', 35)]
[12:02:37] [Server thread/INFO]: SurestTexas00 lost connection: TranslatableComponent{ke
y='disconnect.genericReason', args=[Internal Exception: java.io.IOException: An existing
 connection was forcibly closed by the remote host], siblings=[], style=Style{hasParent=
false, color=null, bold=null, italic=null, underlined=null, obfuscated=null, clickEvent=
null, hoverEvent=null, insertion=null}}
[12:02:37] [Server thread/INFO]: ºbSurestTexas00ºr left the game
[12:02:37] [Server thread/INFO]: That player cannot be found
Failed to grab packet (CLIENT):
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 560, in handle
    id, original = self.packet.grabPacket()
  File "Wrapper\proxy.py", line 1006, in grabPacket
    dataLength = self.unpack_varInt()
  File "Wrapper\proxy.py", line 1030, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "Wrapper\proxy.py", line 1180, in recv
    d = self.socket.recv(length)
  File "C:\Python27\lib\socket.py", line 170, in _dummy
    raise error(EBADF, 'Bad file descriptor')
error: [Errno 9] Bad file descriptor

[12:02:52] [Wrapper.py/INFO] SurestTexas00 logged in (UUID: 4f953255-6775-4dc6-a612-fb42
30588eff | IP: 68.50.227.172)
[12:02:52] [Server thread/INFO]: SurestTexas00[/127.0.0.1:64627] logged in with entity i
d 27896 at (1593.3964577777137, 69.0, 464.64614363203304)
[12:02:52] [Server thread/INFO]: ºbSurestTexas00ºr joined the game
Error in the Server->Client method:
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 969, in handle
    if self.parse(id, original) and self.safe:
  File "Wrapper\proxy.py", line 916, in parse
    properties = client.properties
AttributeError: 'UUID' object has no attribute 'properties'

Last packet IDs (Server->Client) before disconnection:
[('0x39', 10), ('0x9', 2), ('0x37', 52), ('0x3e', 56), ('0x3e', 62), ('0x3e', 63), ('0x3
e', 48), ('0x3e', 61), ('0x38', 37), ('0x38', 35)]
[12:02:52] [Server thread/INFO]: SurestTexas00 lost connection: TranslatableComponent{ke
y='disconnect.genericReason', args=[Internal Exception: java.io.IOException: An existing
 connection was forcibly closed by the remote host], siblings=[], style=Style{hasParent=
false, color=null, bold=null, italic=null, underlined=null, obfuscated=null, clickEvent=
null, hoverEvent=null, insertion=null}}
[12:02:52] [Server thread/INFO]: ºbSurestTexas00ºr left the game
Failed to grab packet (CLIENT):
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 560, in handle
    id, original = self.packet.grabPacket()
  File "Wrapper\proxy.py", line 1006, in grabPacket
    dataLength = self.unpack_varInt()
  File "Wrapper\proxy.py", line 1030, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "Wrapper\proxy.py", line 1180, in recv
    d = self.socket.recv(length)
  File "C:\Python27\lib\socket.py", line 170, in _dummy
    raise error(EBADF, 'Bad file descriptor')
error: [Errno 9] Bad file descriptor

```

**Comment by suresttexas00** [comment]
This is part of the UUID issue

**Comment by suresttexas00** [comment]
This is not the UUID issue, this is a method that returns an inconsistent value:
```
	def getClientByServerUUID(self, id):
		for client in self.clients:
			if str(client.serverUUID) == str(id):
				self.uuidTranslate[str(id)] = str(client.uuid)
				return client
		# if str(id) in self.uuidTranslate:
		# 	return uuid.UUID(hex=self.uuidTranslate[str(id)])
		return False  # no client
```

I commented out the problem lines, which instead of returning the client instance, return the UUID object.. yuck!  UUID object has to "properties", ergo, our error.  I modified this to just return a False if the client is not found... and make sure our code is checking elsewhere for a returned False before attempting to use the client.

**Comment by suresttexas00** [comment]
This will be fixed in the next pull (not #300).  The fix is auto-acceptance of the EULA.
